_gsocproposals/2019/proposal_GlobalModulesIndex.md

```diff
@@ -0,0 +1,41 @@
+---
+title: Implement a GlobalModulesIndex in ROOT and Cling
+layout: gsoc_proposal
+project: ROOT
+year: 2019
+organization:
+  - CERN
+  - University of Lincoln-Nebraska
+---
+
+## Description
+This project aims to implement a GlobalModulesIndex [1],[2], which is expected to speed up ROOT’s [3] startup time by three times. ROOT has own C++ interpreter Cling [4] for getting the reflection information.
+
+We have been implementing C++ modules support in ROOT for the last few years, and we have recently reached the production level. However, there is still a room for optimization, and one of the possible solutions could be the implementation of GlobalModulesIndex. This project will give the selected student an opportunity to contribute to two large and widely-used open source projects: ROOT and Clang/LLVM. Moreover, the student will get a deep knowledge of C++ compiler and interpreter technology throughout this project.
```

Comment:
Since `Clang/LLVM` is mentioned here, should we also add a link to some references?